File: packages/android/formbricksSDK/src/main/java/com/formbricks/formbrickssdk/model/javascript/JsResponseMessage.kt
package com.formbricks.formbrickssdk.model.javascript

import com.formbricks.formbrickssdk.model.response.EventType
import com.formbricks.formbrickssdk.model.response.SurveyResponse
import com.google.gson.Gson
import com.google.gson.annotations.SerializedName

data class JsResponseMessage(
    @SerializedName("event") val event: EventType,
    @SerializedName("responseUpdate") val surveyResponse: SurveyResponse?
) {
    companion object {
        fun from(string: String): JsResponseMessage {
            return Gson().fromJson(string, JsResponseMessage::class.java)
        }
    }
}

File: packages/ios/FormbricksSDK/FormbricksSDK/Model/Javascript/JsResponseMessage.swift
struct JsResponseMessage: Codable {
    let event: EventType
    let responseUpdate: ResponseUpdate?
}

struct ResponseUpdate: Codable {
    let data: [String: AnyCodable]?
    let ttc: [String: AnyCodable]?
    let finished: Bool?
    let variables: [String: AnyCodable]?
    let language: String?
}
